File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/dto/ProyectoSeguimientoEjecucionEconomica.java
package org.crue.hercules.sgi.csp.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Set;

import org.crue.hercules.sgi.csp.model.ConvocatoriaTitulo;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProyectoSeguimientoEjecucionEconomica {

  private Long id;
  private Long proyectoId;
  private String proyectoSgeRef;
  private String nombre;
  private String codigoExterno;
  private Instant fechaInicio;
  private Instant fechaFin;
  private Instant fechaFinDefinitiva;
  private Set<ConvocatoriaTitulo> tituloConvocatoria;
  private BigDecimal importeConcedido;
  private BigDecimal importeConcedidoCostesIndirectos;
}

File: sgi-webapp/src/app/core/models/csp/proyecto-seguimiento-ejecucion-economica.ts
import { I18nFieldValue } from '@core/i18n/i18n-field';
import { DateTime } from 'luxon';

export interface IProyectoSeguimientoEjecucionEconomica {
  /** ProyectoProyectoSgeId */
  id: number;
  proyectoId: number;
  proyectoSgeRef: string;
  nombre: I18nFieldValue[];
  codigoExterno: string;
  fechaInicio: DateTime;
  fechaFin: DateTime;
  fechaFinDefinitiva: DateTime;
  tituloConvocatoria: I18nFieldValue[];
  importeConcedido: number;
  importeConcedidoCostesIndirectos: number;
}
